akka-http-core/src/main/scala/akka/http/impl/engine/parsing/BodyPartParser.scala

```diff
@@ -290,12 +291,11 @@ private[http] object BodyPartParser {
   }
 
   // TODO: load from config
-  val defaultSettings = Settings(
-    maxHeaderNameLength = 64,
-    maxHeaderValueLength = 8192,
-    maxHeaderCount = 64,
-    illegalHeaderWarnings = true,
-    headerValueCacheLimit = 8,
-    uriParsingMode = Uri.ParsingMode.Relaxed,
-    cookieParsingMode = ParserSettings.CookieParsingMode.RFC6265)
+
+  // TODO: load from config
+  object Settings {
+    def apply(s: ParserSettings): Settings = Settings(s.maxHeaderNameLength, s.maxHeaderValueLength, s.maxHeaderCount, s.illegalHeaderWarnings, s.headerValueCacheLimit("default"), s.uriParsingMode, s.cookieParsingMode)
+  }
+
+  val defaultSettings = Settings(ParserSettings.fromSubConfig(ConfigFactory.load, ConfigFactory.systemProperties()))
```

Comment:
This one we'll want to look like this:

```scala
private[http] final class BodyPartParser(defaultContentType: ContentType,
                                         boundary: String,
                                         log: LoggingAdapter,
                                         settings: BodyPartParser.Settings)
```

And then follow the types - see where we need to pass it in, and obtain it from the ActorSystem's config :)